### PR TITLE
feat: FastAPI Implementation of Sotopia Part Two (w websocket)

### PR DESCRIPTION
**Pull request by XuhuiZhou**
Closes # 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed
- [ ] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [ ] Ready for code review

### CODE REVIEW

**Comment by bugsz** [comment]
A simple usage of this api, still working on it: https://github.com/sotopia-lab/sotopia-demo/pull/14

File: pyproject.toml [hunk 1]
```diff
@@ -37,7 +37,12 @@ groq = ["groq"]
 cohere = ["cohere"]
 google-generativeai = ["google-generativeai"]
 examples = ["transformers", "datasets", "scipy", "torch", "pandas"]
-chat = ["fastapi"]
+chat = [
+    "fastapi",
+    "websockets>=13.1",
+    "rel",
```
Comment (ProKil) [inline pyproject.toml:43]: FastAPI and uvicorn make sense.

Can we use `aiohttp` instead of `websockets`? `rel` seems to be not used anywhere

**Review by ProKil — CHANGES_REQUESTED** [review]

File: sotopia/ui/README.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+
+## An example to run simulation with the API
+
+**Get all scenarios**:
+```bash
+curl -X GET "http://localhost:8000/scenarios"
+```
+
+Randomly select a scenario, e.g., `01HZRGTG1K4YQ2CBS9SNH28R9S`
```
Comment (ProKil) [inline sotopia/ui/README.md:186]: #### GET /scenarios

According to the doc:

```
Get all scenarios.

returns:
- scenarios: list[EnvironmentProfile]
```

It should not be a random scenario?

**Review by ProKil — CHANGES_REQUESTED** [review]

**Comment by ProKil** [comment]
Please merge main into this branch.

File: sotopia/ui/README.md (hunk 2 above)
Comment (XuhuiZhou) [reply sotopia/ui/README.md:186]: typo

File: sotopia/ui/README.md [hunk 3]
```diff
@@ -47,6 +47,13 @@ returns:
 - agents: list[AgentProfile]
 
 
+#### GET /episodes
```
Comment (XuhuiZhou) [inline sotopia/ui/README.md:50]: remove this, we don't have this now

**Comment by ProKil** [comment]
> Attention: Patch coverage is 40.17857% with 201 lines in your changes missing coverage. 

Could you fix this?

**Comment by XuhuiZhou** [comment]
> > Attention: Patch coverage is 40.17857% with 201 lines in your changes missing coverage.
> 
> Could you fix this?

okay, let's not worry about this for now in light of the timeline

**Comment by ProKil** [comment]
Since this PR is very large and not covered by CI tests, it is hard to take it directly into the main branch, i.e. I could not read every line to make sure it doesn't break anything or rely on the test cases to make sure the features introduced will not be broken.

To accelerate the development process, I changed the base to a different branch `demo`. Please merge related improvements to that branch, and we will merge it whenever it has been thoroughly tested.

**Pull request merged**